core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -916,4 +920,31 @@ private Cache<TableOperations, FileIO> newFileIOCloser() {
                 })
         .build();
   }
+
+  @Override
+  public void commitTransaction(SessionContext context, List<TableCommit> commits) {
+    ImmutableCommitTransactionRequest.Builder builder = ImmutableCommitTransactionRequest.builder();
+
+    for (TableCommit commit : commits) {
+      UpdateTableRequest.Builder updateTableBuilder = UpdateTableRequest.builderFor(commit.base());
+      commit.changes().forEach(updateTableBuilder::update);
+      UpdateTableRequest updateTableRequest = updateTableBuilder.build();
```

Comment:
Looks like we may need to refactor how we produce requirements. This would not work for non-REST catalogs.